python/modules/IcePy/Communicator.cpp

```diff
@@ -152,6 +153,8 @@ communicatorInit(CommunicatorObject* self, PyObject* args, PyObject* /*kwds*/)
         // We always supply our own implementation of ValueFactoryManager.
         data.valueFactoryManager = ValueFactoryManager::create();
 
+        // data.sliceLoader remains null as we don't want to change the Slice loader for the Ice C++ runtime.
```

Comment:
It actually breaks for Ice::ObjectNotFoundException, which is created/caught in the location-caching code (all in C++).

Reply:
Why is `Ice::ObjectNotFoundException` loaded using the SliceLoader? This is a dispatch exception not an Slice excpetion.

Reply:
ObjectNotFound is a Slice exception defined here:
https://github.com/zeroc-ice/ice/blob/f5995683fe2184a0c44661dfe9d9868c4234a2d8/slice/Ice/Locator.ice#L27

Not to be confused with ObjectNotExistException.